server/server.go

```diff
@@ -53,6 +54,8 @@ type Server struct {
 	routinesPool                  *safe.Pool
 	leadership                    *cluster.Leadership
 	defaultForwardingRoundTripper http.RoundTripper
+	metricsRegistry               metrics.Registry
+	metricsEnabled                bool
```

Comment:
I'm not fan of adding this flag. Can't we remove it and use `metricsRegistry` instead?